Uses a different folder if on a DMM world

RobinDson opened the pull request:
Checks your current world if it is a DMM world or not and changes directory.
Like such: 
![](https://i.imgur.com/MJOqq1G.png)

Closes: #2050

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java:
```diff
@@ -467,10 +468,36 @@ void takeScreenshot(String fileName, boolean displayDate)
 
+			try
+			{
+				World world = new WorldClient().lookupWorlds().findWorld(client.getWorld());
```
This should definitely be determined upon login and not here.

RobinDson replied:
You are correct and this have been adjusted.

SoyChai commented:
Won't this still overwrite level-ups from previous Deadman seasonals? I think we should add some additional timestamping while playing in this mode.

RobinDson commented:
Could add date to all screenshots, just like manual screenshots have.

Adam- commented on runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java:
```diff
@@ -358,10 +360,35 @@ public void hideWidgets(WidgetHiddenChanged event)
+	{
+		try
+		{
+			World world = new WorldClient().lookupWorlds().findWorld(client.getWorld());
```
At the very least, the worlds should only be loaded at plugin start. I think the xptracker plugin also does the same type of thing, maybe a client-wide world service would be a good thing to have. Regardless this is called on the same thread so you never want to do webrequests from it like this.

Adam- commented on runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java:
```diff
@@ -358,10 +360,35 @@ public void hideWidgets(WidgetHiddenChanged event)
 		{
 			return;
 		}
+		fileName = fileName + "";
```
?

SoyChai replied:
Kinda looks like a null safeguard for the filename string, but it's not necessary with the null check right above it.

devLotto commented on runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java:
```diff
@@ -362,6 +377,31 @@ public void hideWidgets(WidgetHiddenChanged event)
+			{
+				IS_DMM_WORLD = false;
+			}
+		}
```
I think everything inside the try { } can be replaced with:

```java
World world = worldResult.findWorld(currentWorld);

boolean deadman = world.getTypes().contains(WorldType.DEADMAN);
boolean seasonalDeadman = world.getTypes().contains(WorldType.SEASONAL_DEADMAN);

IS_DMM_WORLD = deadman || seasonalDeadman;
```

RobinDson replied:
Rewrote it with your code and seem to be working so I'll update it.

devLotto commented:
From Discord:

![](https://i.imgur.com/pEVcDoT.png)

Maybe a good idea

SoyChai commented:
^ This, plus a `{timestamp}-{level up}` filename would work well imo.

RobinDson commented:
Changed the folder name as suggested and also added the option to add a timestamp on each levelup, clue scroll etc.
![image](https://user-images.githubusercontent.com/37742094/39633131-5700b890-4fb7-11e8-8d77-8dfcfc817cd1.png)

Nightfirecat commented:
@deathbeam OP deleted their GitHub account, which leads me to believe they will not be completing this PR. Close?
